core/src/components/nav-link/nav-link.tsx

```diff
@@ -34,7 +34,14 @@ export class NavLink implements ComponentInterface {
   @Prop() routerAnimation?: AnimationBuilder;
 
   private onClick = () => {
-    return navLink(this.el, this.routerDirection, this.component, this.componentProps, this.routerAnimation);
+    const { el } = this;
+    const button = el.querySelector<HTMLIonButtonElement>('ion-button');
```

Comment:
I don't think this is the correct solution since this bug could happen with any component, not just `ion-button`. For example, the following would still reproduce the issue:

```html
<ion-nav-link>
  <button disabled>Hello</button>
</ion-nav-link>
```

We could potentially look into adding a `disabled` property on `ion-nav-link`, though maybe devs could conditionally set the `component` prop based on some application state.

I also checked Vue Router, and disabling a child button does not prevent `<router-link>` from functioning.

Reply:
This example:
```html
<ion-nav-link>
  <button disabled>Hello</button>
</ion-nav-link>
```

Strangely enough does not reproduce the issue. The click event is not propagated to the `ion-nav-link` when clicking a disabled HTML button. This same behavior doesn't occur with a disabled button in the shadow DOM though.

https://github.com/ionic-team/ionic-framework/assets/13732623/61aa3e0c-b8bf-4696-b87f-da4a529e709e

> I also checked Vue Router, and disabling a child button does not prevent <router-link> from functioning.

Would that suggest we shouldn't handle accounting for disabled interactive buttons from `ion-nav-link`?

Reply:
The issue doesn't reproduce because you're clicking where `ion-nav-link` overlaps with `button`. The two don't perfectly overlap, so if click on that area then the issue will reproduce:


https://github.com/ionic-team/ionic-framework/assets/2721089/d966b625-6ed4-4b44-84a7-25155a6a91df

Reply:
In this case, developers could probably conditionally set the `component` property. If they want `ion-nav-link` to do nothing then they can pass `undefined` to `component`

Reply:
We'll want to test this though -- I think keeping the `onClick` handler around will still cause screen readers to read this as "clickable", so we might want to set the `onClick` listener only if `component` is set